### DISCUSSION TITLE
Validation rules required on every model attribute -- even nested json column values

### DISCUSSION THREAD

**Discussion by wdavis** [livewire/livewire#1510] [4. General]
Currently it looks like Livewire v2 requires validation rules to be explicitly set for every attribute on a model. This makes sense if you know exactly what you are dealing with. However a side affect of this is that anything in a json column needs to have a rule defined for it too.

Wildcards keys like `model.items.*.value` cannot be used in rules either -- Lack of an explicit validation rule like `model.items.0.value` end in a `CannotBindToModelDataWithoutValidationRuleException`.

Primitive Example: You have a model with a json column called `items` that has an array of objects with attributes.

You want to store some dynamic bits of information here and allow these to be added/removed in a form wired up like so:

```blade
@foreach($model->items as $item)
    <input wire:model="model.items.{{ $loop->index }}.value" type="text">
@endforeach
```

When updating the value of the input you'll receive:

`Cannot bind property [model.items.0.value] without a validation rule present in the [$rules] array on Livewire component.`

Wildcard support would make this easier to deal with, however I feel like a some flexibility is being lost by having validation rules be a requirement enforced by Livewire.

Thoughts?

**Comment by roni-estein** [comment]
Caleb is waiting for us to whine about it enough. Whine more

**Reply by nuernbergerA** [reply]
That's indeed a general rule :rofl:

**Comment by nuernbergerA** [comment]
Validation `$rules` with an asterisk should be [supported](https://github.com/livewire/livewire/pull/1445) and [tested](https://github.com/livewire/livewire/blob/b994138d2afa611ef3ed7a495b0fa4de4ec25930/tests/Unit/EloquentModelValidationTest.php#L80)

It might be that some last-minute changes have broken something.

@wdavis please open an issue with domain free code to reproduce your issue.

Cheers Adrian
